packages/babel-plugin/src/index.js

```diff
@@ -335,10 +346,17 @@ function processStylexRules(
     return '';
   }
 
+  const constsMap: Map<string, string | number> = new Map();
+  for (const [keyhash, ruleObj] of rules) {
+    if (ruleObj?.constKey && ruleObj?.constVal) {
+      constsMap.set(`var(--${keyhash})`, ruleObj.constVal);
+    }
+  }
+
```

Comment:
Do a filter to separate the rules that defineConsts from the rest of the rules. This way `sortedRules` should not contain any constants.

Also, You'd need to recursively check the values of the constants to replace any constant that references another constant.